library/hmac_drbg.c

```diff
@@ -257,13 +270,19 @@ static int hmac_drbg_reseed_core( mbedtls_hmac_drbg_context *ctx,
         goto exit;
 
     /* 3. Reset reseed_counter */
+    reseed_counter_backup = ctx->reseed_counter;
     ctx->reseed_counter = 1;
 
 exit:
 
     /* 4. Done */
     mbedtls_platform_zeroize( seed, seedlen );
 
+    if( additional_dup != additional || len_dup != len )
+    {
+        return MBEDTLS_ERR_PLATFORM_FAULT_DETECTED;
```

Comment:
@AndrzejKurek ,we should revert reseed counter if we have FI here

Reply:
Added.